packages/eslint-plugin-pf-codemods/lib/rules/v5/datalist-remove-selectableRow.js

```diff
@@ -1,25 +1,244 @@
-const { getPackageImports} = require('../../helpers');
+const {
+  getPackageImports,
+  findVariableDeclaration,
+  getAllJSXElements,
+} = require("../../helpers");
 
 //https://github.com/patternfly/patternfly-react/pull/8827
 module.exports = {
-  meta: {},
+  meta: { fixable: "code" },
   create: function (context) {
-    const dataListImport =  getPackageImports(context, '@patternfly/react-core')
-      .filter(specifier => specifier.imported.name == 'DataList');
+    const dataListImport = getPackageImports(
+      context,
+      "@patternfly/react-core"
+    ).find((specifier) => specifier.imported.name == "DataList");
 
-    return dataListImport.length === 0 ? {} : {
-      JSXOpeningElement(node) {
-        if (dataListImport.map(imp => imp.local.name).includes(node.name.name)) {
-          const selectableRow = node.attributes.find(a => a.name?.name === 'selectableRow');
-          if (selectableRow) {
-            context.report({
-              node,
-              message: `DataList's selectableRow property has been replaced with onSelectableRowChange. The order of the params in the callback has also been updated so that the event param is first.`,
-            });
-          }
+    if (!dataListImport) {
+      return {};
+    }
+
+    let rangeStartsToReplaceConst = [];
+
+    const sc = context.getSourceCode();
+
+    const getObjectProp = (expr, propName) =>
+      expr.properties.find(
+        (prop) =>
+          (prop.key.type === "Identifier" && prop.key.name === propName) ||
+          (prop.key.type === "Literal" && prop.key.value === propName)
+      );
+
+    const swapCallbackParam = (params, toReplace) => {
```

Comment:
Good idea, that didn't come to my mind, that we can run some rules before others. This rule can be a good case for separating the callback swap into another rule. Because currently I use not only `swapCallbackParam`, but also some not very nice string handling in `handleObjectExpression`, where instead of just replacing object like this ` { onChange: function (id, event) {} };` with the `function (id, event) {}`, I am swapping the params right away, which needs to be done with the not so nice combination of strings.

Working on a refactoring right now to make the swapping as a separate rule.